lib/basicauth.php

```diff
@@ -24,6 +24,31 @@ public function __construct($authUrl) {
 	 * @return true/false
 	 */
 	public function checkPassword($uid, $password) {
+		/*
+		 * Connect without user/name password to make sure
+		 * URL is indeed authenticating or not...
+		 */
+		stream_context_set_default(array(
+		  'http'=>array(
+		    'method'=>"GET",
+		  ))
+		);
+		$canary = get_headers($this->authUrl, 1);
+		if(!$canary) {
+			OC::$server->getLogger()->error(
+				'ERROR: Not possible to connect to BasicAuth Url: '.$this->authUrl,
+				['app' => 'user_external']
+			);
+			return false;
+		}
+		if (!isset($canary['WWW-Authenticate'])) {
```

Comment:
That should do the trick:

if (!isset(array_change_key_case($headers, CASE_LOWER)['www-authenticate'])) {
...

Reply:
yes, that's probably the better way than looping...

Reply:
Adjusted the check so that it is case insensitve.